vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts

```diff
@@ -0,0 +1,308 @@
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import * as https from 'https';
+
+import { FileUtilities } from '../Utils/FileUtilities';
+import { IGlobalInstaller } from './IGlobalInstaller';
+import { IAcquisitionWorkerContext } from './IAcquisitionWorkerContext';
+import { VersionResolver } from './VersionResolver';
+import { DotnetConflictingGlobalWindowsInstallError, DotnetCustomLinuxInstallExistsError, DotnetUnexpectedInstallerOSError } from '../EventStream/EventStreamEvents';
+import { ICommandExecutor } from '../Utils/ICommandExecutor';
+import { CommandExecutor } from '../Utils/CommandExecutor';
+/* tslint:disable:only-arrow-functions */
+/* tslint:disable:no-empty */
+
+/**
+ * @remarks
+ * This class manages global .NET SDK installations for windows and mac.
+ * Both of these OS's have official installers that we can download and run on the machine.
+ * Since Linux does not, it is delegated into its own set of classes.
+ */
+export class WinMacGlobalInstaller extends IGlobalInstaller {
+
+    private installerUrl : string;
+    private installingVersion : string;
+    protected commandRunner : ICommandExecutor;
+    public cleanupInstallFiles = true;
+
+    constructor(context : IAcquisitionWorkerContext, installingVersion : string, installerUrl : string, executor : ICommandExecutor | null = null)
+    {
+        super(context);
+        this.installerUrl = installerUrl
+        this.installingVersion = installingVersion;
+        this.commandRunner = executor ?? new CommandExecutor();
+    }
+
+    public async installSDK(): Promise<string>
+    {
+        // Check for conflicting windows installs
+        if(os.platform() === 'win32')
+        {
+            const conflictingVersion = await this.GlobalWindowsInstallWithConflictingVersionAlreadyExists(this.installingVersion);
+            if(conflictingVersion !== '')
+            {
+                if(conflictingVersion === this.installingVersion)
+                {
+                    // The install already exists, we can just exit with Ok.
+                    return '0';
+                }
+                const err = new DotnetConflictingGlobalWindowsInstallError(new Error(`An global install is already on the machine: version ${conflictingVersion}, that conflicts with the requested version.
+                    Please uninstall this version first if you would like to continue.`));
+                this.acquisitionContext.eventStream.post(err);
+                throw err;
+            }
+        }
+
+        const installerFile : string = await this.downloadInstaller(this.installerUrl);
+        const installerResult : string = await this.executeInstall(installerFile);
+
+        if(this.cleanupInstallFiles)
+        {
+            FileUtilities.wipeDirectory(path.dirname(installerFile));
+        }
+
+        const validInstallerStatusCodes = ['0', '1641', '3010']; // Ok, Pending Update, + Update Starting Now
```

Comment:
These are reboot codes - they don't have anything to do with updates, ERROR_SUCCESS, ERROR_SUCCESS_REBOOT_REQUIRED, ERROR_SUCCESS_REBOOT_INITIATED

Reply:
Whoops, thanks. For some reason I merged concept of installer reboots and updates in my head.